site/tests/integration/blockpage.test.ts

```diff
@@ -0,0 +1,113 @@
+import { expect, test } from "playwright-test-coverage";
+
+import blocksData from "../../blocks-data.json";
+
+const codeBlock = blocksData.find(
+  ({ name }) => name === "@hashintel/block-code",
+);
+
+const unsupportedBlock = blocksData.find(
+  ({ name }) => name === "@hashintel/block-embed",
+);
+
+if (!codeBlock || !unsupportedBlock) {
+  throw new Error("Code and Embed blocks need to be prepared before tests run");
+}
+
+test("Block page should contain key elements", async ({ page, isMobile }) => {
+  await page.goto(codeBlock.blockPackagePath);
+
+  await expect(
+    page.locator(`h1:has-text('${codeBlock.displayName}')`),
+  ).toBeVisible();
+
+  await expect(
+    page.locator(`p:has-text('${codeBlock.description}')`),
+  ).toBeVisible();
+
+  await expect(page.locator(`text=@${codeBlock.author}`)).toBeVisible();
+  await expect(page.locator(`text=@${codeBlock.author}`)).toHaveAttribute(
+    "href",
+    `/@${codeBlock.author}`,
+  );
+  await expect(page.locator(`text=V${codeBlock.version}`)).toBeVisible();
+
+  const blockExample = codeBlock.examples[0] as Record<string, string>;
+
+  if (isMobile) {
+    await page.locator("text=Source Code").click();
+    await expect(page.locator("text=Data Source")).toBeVisible();
+  } else {
+    await expect(page.locator("text=Block Properties")).toBeVisible();
+  }
+  await expect(page.locator("#simple-tabpanel-0 >> textarea")).toHaveValue(
+    JSON.stringify(blockExample, null, 2),
+  );
+
+  if (isMobile) {
+    await page.locator("text=Data Source").click();
+    await page.locator("text=Block Schema").click();
+  } else {
+    await page.locator("text=Block Schema").click();
+  }
+  await expect(page.locator("#simple-tabpanel-1")).toBeVisible();
+  // await expect(page.locator("#simple-tabpanel-1")).not.toBeEmpty();
+
+  if (isMobile) {
+    await page.locator("text=Preview").click();
+  }
+
+  await expect(page.locator("iframe[title='block']")).toBeVisible();
+  await expect(
+    page.frameLocator("iframe[title='block']").locator("input"),
+  ).toHaveValue(blockExample.caption!);
+
+  // check if readme was displayed
+  await expect(page.locator("article")).toBeVisible();
+
+  await expect(page.locator("h2:has-text('Repository')")).toBeVisible();
+
+  await expect(
+    page.locator("a:below(h2:has-text('Repository'))").first(),
+  ).toHaveAttribute("href", codeBlock.repository);
+
+  await expect(page.locator("text=Explore more blocks")).toBeVisible();
+
+  // Block slider
```

Comment:
This comment is a bit like:
<img width="291" alt="Screenshot 2022-07-21 at 19 25 04" src="https://user-images.githubusercontent.com/608862/180286641-77a653a3-2b1a-49a1-9030-e9f77205fd6a.png">

😅 I'd remove all comments that repeat that selectors and ideally use selectors that don't require comments.

Reply:
Agreed. Updated in [5adad49](https://github.com/blockprotocol/blockprotocol/pull/450/commits/5adad49fdf2865ff470f34040be638718ea3a2aa)